drivers/xen/events.c

```diff
@@ -21,6 +21,80 @@ static evtchn_handle_t event_channels[EVTCHN_2L_NR_CHANNELS];
 
 static void empty_callback(void *data) { }
 
+int evtchn_alloc_unbound(domid_t dom, domid_t remote_dom)
+{
+	int rc;
+	struct evtchn_alloc_unbound alloc = {
+		.dom = dom,
+		.remote_dom = remote_dom,
+	};
+
+	rc = HYPERVISOR_event_channel_op(EVTCHNOP_alloc_unbound, &alloc);
+	if (rc == 0) {
+		rc = alloc.port;
+	}
+
+	return rc;
+}
+
+int evtchn_bind_interdomain(domid_t remote_dom, evtchn_port_t remote_port)
+{
+	int rc;
+	struct evtchn_bind_interdomain bind = {
+		.remote_dom = remote_dom,
+		.remote_port = remote_port,
+	};
+
+	rc = HYPERVISOR_event_channel_op(EVTCHNOP_bind_interdomain, &bind);
+	if (rc == 0) {
+		rc = bind.local_port;
+	}
+
+	return rc;
+}
+
+int evtchn_status(evtchn_status_t *status)
+{
+	return HYPERVISOR_event_channel_op(EVTCHNOP_status, status);
+}
+
+int evtchn_unmask(evtchn_port_t port)
+{
+	struct evtchn_unmask unmask = {
+		.port = port,
+	};
+
+	return HYPERVISOR_event_channel_op(EVTCHNOP_unmask, &unmask);
+}
+
+int evtchn_close(evtchn_port_t port)
+{
+	struct evtchn_close close = {
+		.port = port,
+	};
+
+	return HYPERVISOR_event_channel_op(EVTCHNOP_close, &close);
+}
+
+int evtchn_reset(domid_t dom)
```

Comment:
My main goal is to use Zephyr as Domain 0 OS, so I thought it will be useful, but as I can see even Linux don't use this hypercall. So agree, it is redundant, removed.